tokenmagic/module/tokenmagic.js

```diff
@@ -761,6 +841,21 @@ export function TokenMagic() {
 
         var filterById = (filterId != null && typeof filterId === "string");
 
+        function destroyClearedFilters(theFilters) {
+            if (theFilters instanceof Array) {
+                var tmFilters = theFilters.filter(filter =>
+                    filterById
+                        ? filter.hasOwnProperty("filterId") && filter.filterId === filterId
+                        : filter.hasOwnProperty("filterId")
+                );
+
+                for (const filter of tmFilters) {
+                    filter.enabled = false;
+                    filter.destroy();
+                }
+            }
+        };
+

```

Comment:
When a filter is deleted, allow TMFX to properly clean if needed (pixi sprite, texture, etc.)